doc/plugin_tutorial/tuto1/src/g_tuto1.mlg

```diff
@@ -16,141 +16,278 @@ open Stdarg
 
 }
 
-VERNAC COMMAND EXTEND HelloWorld CLASSIFIED AS QUERY
-| [ "Hello" string(s) ] ->
-  { Feedback.msg_notice (strbrk "Hello " ++ str s) }
-END
+(*** Printing inputs ***)
 
-(* reference is allowed as a syntactic entry, but so are all the entries
-   found the signature of module Prim in file coq/parsing/pcoq.mli *)
+(*
+ * This command prints an input from the user.
+ *
+ * A list with allowable inputs can be found in interp/stdarg.mli,
+ * plugin/ltac/extraargs.mli, and plugin/ssr/ssrparser.mli
+ * (remove the wit_ prefix), but not all of these are allowable
+ * (unit and bool, for example, are not usable from within here).
+ *
+ * We include only some examples that are standard and useful for commands.
+ * Some of the omitted examples are useful for tactics.
+ *
+ * Inspector is our own file that defines a simple messaging function.
+ * The printing functions (pr_qualid and so on) are in printing.
+ *
+ * Some of these cases would be ambiguous if we used "What's" for each of
+ * these. For example, all of these are terms. We purposely disambiguate.
+ *)
+VERNAC COMMAND EXTEND WhatIsThis CLASSIFIED AS QUERY
+| [ "What's" constr(e) ] ->
+   {
+     let env = Global.env () in (* we'll explain later *)
+     let sigma = Evd.from_env env in (* we'll explain later *)
+     Inspector.print_input e (Ppconstr.pr_constr_expr env sigma) "term"
+   }
+| [ "What" "kind" "of" "term" "is" string(s) ] ->
```

Comment:
Do we say anywhere that `"foo"` makes `foo` a keyword (such that after loading the plugin it is impossible to define a constant called `kind` for instance)? To avoid making keywords when they're not needed to make the parser work, use `IDENT "foo"`.

Reply:
It does so only inside the `GEXTEND` whing, not inside the `ARGUMENT EXTEND` thing IIRC.

Reply:
Is there any action to take here?

Reply:
I guess not then.